### PR TITLE
[enhancement](cloud) dump error file when footer parse failure

### PR DESCRIPTION
**Pull request by freemandealer**
dump error file to local disk in cloud mode for better debug. limit the error file directory to avoid disk capacity problem.

## Proposed changes

Issue Number: close #xxx

### CODE REVIEW

**Comment by freemandealer** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment.cpp [hunk 1]
```diff
@@ -294,6 +295,41 @@ Status Segment::new_iterator(SchemaSPtr schema, const StorageReadOptions& read_o
...
+        return create_st;
+    }
+    size_t dir_size = 0;
+    RETURN_IF_ERROR(io::global_local_filesystem()->directory_size(dir_path, &dir_size));
+    if (dir_size > config::file_cache_error_log_limit_bytes) {
+        LOG(WARNING) << "error file dir size is too large: " << dir_size;
+        return Status::OK();
+    }
+
+    std::string file_path = dir_path + std::to_string(cur_time) + "_" + ss.str();
```
Comment (gavinchou) [inline be/src/olap/rowset/segment_v2/segment.cpp:324]: use this pattern to get rid of duplicated dumps (check existence before read and write)
```
    std::string file_path = dir_path + this->_rowset_id.to_string() + "_" + segment_id + "_" + offset + "_" + size;
```

**Review by gavinchou — CHANGES_REQUESTED** [review]

**Comment by freemandealer** [comment]
run buildall

**Comment by freemandealer** [comment]
run buildall

**Review by gavinchou — DISMISSED** [review]

**Review by Yukang-Lian — APPROVED** [review]

**Comment by freemandealer** [comment]
run beut

**Comment by freemandealer** [comment]
run buildall

**Comment by freemandealer** [comment]
run buildall

**Comment by freemandealer** [comment]
run beut

**Comment by freemandealer** [comment]
run buildall

**Review by gavinchou — APPROVED** [review]

**Pull request merged**